sdk/pinocchio/src/sysvars/tests/slot_hashes_tests.rs

```diff
@@ -0,0 +1,855 @@
+            Some(&entries[entry_count - 1].1)
+        );
+        assert_eq!(slot_hashes.get_hash(START_SLOT + 1), None);
+
+        // Test empty list explicitly
+        let empty_entries = generate_mock_entries(0, START_SLOT, DecrementStrategy::Strictly1);
+        let empty_data = create_mock_data(&empty_entries);
+        let empty_hashes = unsafe { SlotHashes::new_unchecked(empty_data.as_slice(), 0) };
+        assert_eq!(empty_hashes.get_hash(100), None);
+
+        let pos_start_plus_1 = slot_hashes.position(START_SLOT + 1);
+        assert!(
+            pos_start_plus_1.is_none(),
+            "position(START_SLOT + 1) should be None"
+        );
+    }
+
+    #[test]
+    fn test_basic_getters_and_iterator_no_std() {
+        const NUM_ENTRIES: usize = 512;
+        const START_SLOT: u64 = 2000;
+        let entries = generate_mock_entries(NUM_ENTRIES, START_SLOT, DecrementStrategy::Strictly1);
+        let data = create_mock_data(&entries);
+        let slot_hashes = unsafe { SlotHashes::new_unchecked(data.as_slice(), NUM_ENTRIES) };
+
+        assert_eq!(slot_hashes.len(), NUM_ENTRIES);
+        assert!(!slot_hashes.is_empty());
+
+        let entry0 = slot_hashes.get_entry(0);
+        assert!(entry0.is_some());
+        assert_eq!(entry0.unwrap().slot(), START_SLOT); // Check against start slot
+        assert_eq!(entry0.unwrap().hash, [0u8; HASH_BYTES]); // First generated hash is [0u8; 32]
+
+        let entry2 = slot_hashes.get_entry(NUM_ENTRIES - 1); // Last entry
+        assert!(entry2.is_some());
+        assert_eq!(entry2.unwrap().slot(), entries[NUM_ENTRIES - 1].0);
+        assert_eq!(entry2.unwrap().hash, entries[NUM_ENTRIES - 1].1);
+        assert!(slot_hashes.get_entry(NUM_ENTRIES).is_none()); // Out of bounds
+
+        for (i, entry) in slot_hashes.into_iter().enumerate() {
+            assert_eq!(entry.slot(), entries[i].0);
+            assert_eq!(entry.hash, entries[i].1);
+        }
+        assert!(slot_hashes.into_iter().nth(NUM_ENTRIES).is_none());
+
+        // Test ExactSizeIterator hint
+        let mut iter_hint = slot_hashes.into_iter();
+        assert_eq!(iter_hint.size_hint(), (NUM_ENTRIES, Some(NUM_ENTRIES)));
+        iter_hint.next();
+        assert_eq!(
+            iter_hint.size_hint(),
+            (NUM_ENTRIES - 1, Some(NUM_ENTRIES - 1))
+        );
+        // Skip to end
+        for _ in 1..NUM_ENTRIES {
+            iter_hint.next();
+        }
+        iter_hint.next();
+        assert_eq!(iter_hint.size_hint(), (0, Some(0)));
+
+        // Test empty case
+        let empty_data = create_mock_data(&[]);
+        let empty_hashes = unsafe { SlotHashes::new_unchecked(empty_data.as_slice(), 0) };
+        assert_eq!(empty_hashes.len(), 0);
+        assert!(empty_hashes.is_empty());
+        assert!(empty_hashes.get_entry(0).is_none());
+        assert!(empty_hashes.into_iter().next().is_none());
+    }
+
+    #[test]
+    fn test_get_entry_count_no_std() {
```

Comment:
I might be wrong, but it seems like this is 5 separate tests sequenced one after another, no?